cmd/livepeer/starter/starter.go

```diff
@@ -737,16 +746,26 @@ func StartLivepeer(ctx context.Context, cfg LivepeerConfig) {
 				// Can't divide by 0
 				panic(fmt.Errorf("-pixelsPerUnit must be > 0, provided %d", *cfg.PixelsPerUnit))
 			}
+			pixelsPerUnit := new(big.Rat).SetInt64(int64(*cfg.PixelsPerUnit))
 			if cfg.PricePerUnit == nil {
 				// Prevent orchestrators from unknowingly providing free transcoding
 				panic(fmt.Errorf("-pricePerUnit must be set"))
 			}
-			if *cfg.PricePerUnit < 0 {
-				panic(fmt.Errorf("-pricePerUnit must be >= 0, provided %d", *cfg.PricePerUnit))
+			pricePerUnit, currency, err := parsePricePerUnit(*cfg.PricePerUnit)
+			if err != nil {
+				panic(fmt.Errorf("-pricePerUnit must be a valid integer with an optional currency, provided %v", *cfg.PricePerUnit))
+			} else if pricePerUnit.Sign() < 0 {
+				panic(fmt.Errorf("-pricePerUnit must be >= 0, provided %s", pricePerUnit))
+			}
+			err = startPriceUpdateLoop(ctx, *cfg.EthUrl, *cfg.PriceFeedAddr, pricePerUnit, currency, func(price *big.Rat) {
+				n.SetBasePrice("default", new(big.Rat).Quo(pricePerUnit, pixelsPerUnit))
```

Comment:
The same comment as above. Unfortunately, we do allow dynamic price changes for both B and O. So, we need to covert this as well.